WPMedia/fusion-news-theme-blocks
# TMEDIA-231 : Add an optional setting to confine video heights in bloc…

## Description
Added settings and styling to allow the users of pagebuilder to determine if a video will vertically fit in screen.

## Jira Ticket
- [TMEDIA-231](https://arcpublishing.atlassian.net/browse/TMEDIA-231)

## Acceptance Criteria
* When videos are within the article body, the video should be entirely visible within the viewport, like the example screenshot shows above. 
* The ratio of the video should be kept the same, and the video should be centered horizontally within the article body. 
* Text should not wrap around the video. 
* Tests and documentation are updated as necessary

## Test Steps

* Checkout this branch `git checkout TMEDIA-231-video-height-constraints`
* Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/engine-theme-sdk,@wpmedia/article-body-block,@wpmedia/extra-large-promo-block,@wpmedia/large-promo-block,@wpmedia/lead-art-block,@wpmedia/video-player-block,@wpmedia/video-promo-block,@wpmedia/top-table-list-block`

Testing can be done on the following:

video-player-block / top-table-list-block / extra-large-promo-block / large-promo-block
pagebuilder/editor/curate?p=p1QhuC8M1z7U89Cgs&v=v57BsowrKn7U89Cgs

video-player-block / video-promo-block
pagebuilder/editor/curate?p=pRHVAE2xawxxHQ2is&v=v9I63ORx3xxHQ2is

Above video elements should all have a “Video Settings” configuration option to allow the enabling of the feature as well as defining a percentage of the viewport to constrain to (within limits)

Shrinking the window vertically will reduce the videos in size to acceptable levels.


## Effect Of Changes
### Before
*Please see ticket for examples of before*

### After
Videos will now scale to fit if the appropriate selection is made by the user:
![image](https://user-images.githubusercontent.com/2287238/119393644-3631fe00-bc9f-11eb-88da-b3941faef2b8.png)

![image](https://user-images.githubusercontent.com/2287238/119393592-287c7880-bc9f-11eb-9547-bb873d115657.png)
![image](https://user-images.githubusercontent.com/2287238/119393705-4a75fb00-bc9f-11eb-9226-e9a9c36fe0e8.png)

## Dependencies or Side Effects
- https://github.com/WPMedia/engine-theme-sdk/pull/349
- Visual Side Effects include black bars on the left and right of the video.

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
